src/components/wizards/create-vm/CreateVMWizard.js

```diff
@@ -0,0 +1,366 @@
+        title: 'Workload Profile',
+        type: 'dropdown',
+        default: '--- Select Workload Profile ---',
+        values: this.props.workloadProfiles,
+        required: true
+      },
+      startVM:{
+        title: 'Start virtual machine on creation',
+        type: 'checkbox'
+      },
+      createTemplate:{
+        title: 'Create new teplate from configuration',
+        type: 'checkbox'
+      },
+      cloudInit:{
+        title: 'Use cloud-init',
+        type: 'checkbox',
+      },
+      hostName:{
+        title: 'Hostname',
+        type: 'text',
+        isVisible: (basicSettings) => _.get(basicSettings, 'cloudInit.value', false),
+        required: true
+      },
+      authKeys:{
+        title: 'Authenticated SSH Keys',
+        type: 'textarea',
+        isVisible: (basicSettings) => _.get(basicSettings, 'cloudInit.value', false),
+        required: true
+      }
+  
+    }
+  
+    wizardStepsNewVM = [
+      {
+        title: 'Basic Settings',
+        render: () => <Form horizontal>
+          {Object.keys(this.basicSettings).map(key => {
+            if (this.basicSettings[key].isVisible && this.basicSettings[key].isVisible(this.state.basicSettings) === false){
+              return null;
+            }
+            let child = null;
+            const valid = _.get(this.state.basicSettings[key], 'valid', null);
+            if (this.basicSettings[key].type === 'text') {
+              child = <FormControl type="text" value={_.get(this.state.basicSettings[key], 'value', '')} onChange={event => this.onFormChange(event.target.value, key)} />;
+            }
+            if (this.basicSettings[key].type === 'textarea'){
+              child = <FormControl componentClass="textarea" value={_.get(this.state.basicSettings[key], 'value', '')} onChange={event => this.onFormChange(event.target.value, key)} />;
+            }
+            if (this.basicSettings[key].type === 'dropdown'){
+              child = <ButtonGroup justified>
+                <DropdownButton
+                  id={`wizard-dropdown-${key}`}
+                  bsStyle="default"
+                  className="wizard-dropdown"
+                  title={_.has(this.state.basicSettings[key], 'value') ? this.state.basicSettings[key].value : this.basicSettings[key].default}
+                  onSelect={value => this.onFormChange(value, key)}>
+                  {this.basicSettings[key].values.map(value => <MenuItem key={value.name} eventKey={value.value ? value.value : value.name}>{value.name}</MenuItem>)}
+                </DropdownButton>
+              </ButtonGroup>;
+            }
+            if (this.basicSettings[key].type === 'checkbox') {
+              child = <Checkbox checked={_.get(this.state.basicSettings[key], 'value', false)} onChange={event => this.onFormChange(event.target.checked, key)}>{this.basicSettings[key].title}</Checkbox>;
+            }
+            return <FormGroup key={key} validationState={valid ? 'error' : null}>
```

Comment:
please use brackets () and put FormGroup on the newline. IMO it is more readable because of the same spacing

Reply:
+ applies also elsewhere

Reply:
done, hopefully everywhere

Reply:
Yes, this is also the syntax form preferred by patternfly-react project.